units/src/parse.rs

```diff
@@ -78,6 +78,10 @@ mod sealed {
 ///
 /// If the caller owns `String` or `Box<str>` which is not used later it's better to pass it as
 /// owned since it avoids allocation in error case.
+///
+/// # Panics
+///
+/// Panics if the type is larger than 255 bits.
```

Comment:
The compiler gave you a dud steer on this one mate. This call to `expect` can never happen (that is why its ok to use `expect`).

We can just allow the lint on this function.

Reply:
Oh maybe I'm wrong. I didn't see @apoelstra comment above before I reviewed.

Reply:
I think I'm right here, this can't panic because of the `Integer` trait (sealed private trait which is only implemented for its upto and including `u28`.

Reply:
Ah, ok, I missed (i.e. didn't bother checking) if the trait was sealed.

Reply:
I have changed to to just allow and not have the added doc section.